### PR TITLE
improve the Python wrapper

### PR DESCRIPTION
**Pull request by yangyushi**
I made several improvements

1. correctly implemented multiple frame analysis (again). The previous issue is that the output mode is 'w' so only the last frame will be analysed.
2. If the number of analyse frame is given by the user, then the user input have a higher priority
3. I re-implemented the function to get particle-level  information of the TCC clusters.

The minimum example to access the new function is,

```python
import numpy as np
from tcc_python_scripts.tcc import wrapper

coords = np.random.random((25, 1000, 3))
box = [1, 1, 1]

tcc = wrapper.TCCWrapper()
# only the first 10 out of 25 frames will be analysed
tcc.input_parameters['Run']['frames'] = 10
tcc.input_parameters['Output']['Raw'] = True
tcc.input_parameters['Simulation']['rcutAA'] = 0.2
tcc.clusters_to_analyse = ['sp5a', 'sp4a', 'sp3a']
tcc.run(box, coords)

# print first frame of sp5a, particles 1 - 5
cluster_dict = tcc.get_cluster_dict(tcc.clusters_to_analyse)
print("sp5a in the 1st frame: \n", cluster_dict['sp5a'][0][:5].T)

# print all clusters in the first frame
print("\nAll cluster in the 1st frame")
tables = tcc.get_cluster_table(tcc.clusters_to_analyse)
print(tables[0].head())
```

with the following result,

```
sp5a in the 1st frame: 
 [[ True  True False  True  True]]

All cluster in the 1st frame
    sp5a  sp4a   sp3a
0   True  True  False
1   True  True   True
2  False  True  False
3   True  True   True
4   True  True   True
```

### CODE REVIEW

File: tcc_python_scripts/tcc/wrapper.py [hunk 1]
```diff
@@ -303,3 +307,134 @@ def active_clusters(self):
...
+        ..code-block::
+
+            id, FCC, 13A, 12E, 11F, 10B
+            1,    0,   0,   0,   0,   1   # particle 1 is in 10B
+            2,    1,   0,   0,   0,   0   # particle 2 is in FCC
+            3,    0,   0,   0,   0,   0   # particle 3 is not in any cluster
+            ...
+
+        For one particle, this function count all the types of clusters that this
+            particle belongs to. The result corresponds to the gross population.
```
Comment (tranqui) [inline tcc_python_scripts/tcc/wrapper.py:410]: What does this latter paragraph in the documentation mean?

**Review by tranqui — COMMENTED** [review]
Great work Yushi. I've been through your code and it looks good. My only query pertains to the function `get_cluster_table` because I wasn't 100% over how it functions. If you pass it multiple frames, does it return a per-particle count of the number of frames that the particle is in the particular cluster? Or is it always a boolean table?

Also, I think it would be helpful to attach the example script from your post above as an example inside the git repository.

In general though, fantastic work Yushi :)

File: tcc_python_scripts/tcc/wrapper.py (hunk 1 above)
Comment (yangyushi) [reply tcc_python_scripts/tcc/wrapper.py:410]: I was trying to claim that this is not the "net" counting, but this description is a bit redundant and confusing.

**Review by yangyushi — COMMENTED** [review]
Hi Josh the `get_cluster_table` returns many Boolean tables for different frames. I thought about returning a numpy array with shape `(n_frame, n_particle, 3)` but this won't work for experimental data with particle number changing from frame to frame.

**Comment by yangyushi** [comment]
I will improve the doc of the code and find a way to update the pull request later (I don't know how to do it now)

**Comment by tranqui** [comment]
The pull request is in the devel branch, so you should be able to just push your commits there to update the pull request automatically.

File: tcc_python_scripts/file_readers/xyz.py [hunk 2]
```diff
@@ -123,3 +125,104 @@ def write(output_filename, particle_coordinates, species=None):
     """
     snapshot = XYZSnapshot(particle_coordinates, species=species)
     snapshot.write(output_filename)
+
+
+def check_coordinates_type(coordinates):
+    """
+    Determine if a given particle coordinates are multiple frames or
```
Comment (FTurci) [inline tcc_python_scripts/file_readers/xyz.py:132]: I do not understand this: rewrite.

**Review by FTurci — APPROVED** [review]

**Pull request merged**